src/frontend/src/components/deviceListItem.ts

```diff
@@ -1,17 +1,32 @@
 import { TemplateResult, html } from "lit-html";
 import { Setting, settingName } from "../flows/manage/deviceSettings";
-import { warningIcon, dropdownIcon, lockIcon } from "./icons";
+import {
+  warningIcon,
+  errorIcon,
+  successIcon,
+  dropdownIcon,
+  lockIcon,
+} from "./icons";
 
 // A simple representation of "device"s used on the manage page.
 export type Device = {
   // All the settings allowed for a particular device
-  settings: Setting[];
+  settings?: Setting[];
   // The displayed name of a device (not exactly the "alias") because
   // recovery devices handle aliases differently.
-  label: string;
-  isRecovery: boolean;
-  isProtected: boolean;
-  warn?: TemplateResult;
+  label: string | TemplateResult;
+  recovery?: "phrase" | "device";
+  isProtected?: boolean;
```

Comment:
what's `isProtected` undefined? does `undefined` has a different meaning that `false`?

Reply:
@peterpeterparker yeah only phrases can be protected, ok I will separate devices form recovery methods